### ISSUE TITLE
BUG: init_wfc and out_wfc_{basis} function CANNOT cooperate

### ISSUE THREAD

**Issue by QuantumMisaka** [deepmodeling/abacus-develop#3118]
## Introduction

Theoretically, if wavefunction file is output and can be read-in for initial guess of next SCF calculation again, 
the next SCF process will be much more quicker, and the electronic structure should be stable between the two SCF process.

But in ABACUS, something is wrong during :
1. do one SCF with `out_wfc_{basis} 1/2` to generate wavefunction file
2. do the same SCF with `init_wfc file` to read wavefunction file as the initial guess of SCF

## Example: Fe5C2-510-CH+CO+H
A calculation example for Fe-based FTS catalysts surface and its adsorption.
In this system, magnetic moment of Fe and C should around 1.7~1.8 and -0.3, respectively. 

![image](https://github.com/deepmodeling/abacus-develop/assets/43902832/91c3dae2-ca99-4151-839c-24068df49c4a)

[FeCHO_initwfc.tar.gz](https://github.com/deepmodeling/abacus-develop/files/13197513/FeCHO_initwfc.tar.gz)

calculation is done by LCAO basis

### Target: 
do 2 step of SCF, one generate wavefunction file for another to read, 
the next SCF should be much quicker than the first SCF,
with stable magnetic property. 

### Performance
The first step is well, which `AMAG=143, TMAG=156`. 
But when do next THE SAME SCF step by reading wavefunction file, 
The magnetic moment will change, lead to wrong magnetic state and SCF fail to converge.

Above result is from `out_wfc_lcao 1` and `init_wfc file`, which use `.txt` wavefunction file.
When using `out_wfc_lcao 2` to use binary wavefunction file. error occured:
```
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                         NOTICE                           
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
 Can't find the wave function file: LOWF.dat
 CHECK IN FILE : OUT.INITWFC/warning.log
 
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                         NOTICE                           
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

## Example: MgO
MgO example is from Colomb Academy as a model system

[MgO_initwfc.tar.gz](https://github.com/deepmodeling/abacus-develop/files/13197506/MgO_initwfc.tar.gz)

### Target
do 2 step of SCF, one generate wavefunction file for another to read, 
the next SCF should be much quicker than the first SCF

In this example, the wavefunction file read-in performance is tested on ABACUS-LCAO, ABACUS-PW and VASP. 

### Performance
For ABACUS, the only successful job is LCAO and `out_wfc_lcao 1`. 
which give the result in the first SCF:
```
 START CHARGE      : atomic
 DONE(2.02661    SEC) : INIT SCF
 ITER   ETOT(eV)       EDIFF(eV)      DRHO       TIME(s)
 GE1    -7.655079e+03  0.000000e+00   1.146e-01  2.054e+00
 GE2    -7.661793e+03  -6.713720e+00  9.807e-02  1.785e+00
 GE3    -7.665177e+03  -3.383541e+00  1.602e-02  1.810e+00
 GE4    -7.665188e+03  -1.106969e-02  2.253e-03  1.704e+00
 GE5    -7.665189e+03  -1.581676e-03  2.710e-04  1.673e+00
 GE6    -7.665189e+03  -1.030549e-05  2.788e-05  1.732e+00
 GE7    -7.665189e+03  -1.036194e-07  1.339e-06  1.772e+00
 GE8    -7.665189e+03  -3.805100e-10  9.587e-08  1.790e+00
 GE9    -7.665189e+03  -1.392110e-11  1.568e-09  1.662e+00
 GE10   -7.665189e+03  1.546789e-12   1.776e-10  1.762e+00
```
and the result in the next SCF:
```
 WAVEFUN -> CHARGE
 GE1    -7.665189e+03  0.000000e+00   3.068e-03  2.727e+00
 GE2    -7.665189e+03  -1.763684e-06  9.194e-04  1.684e+00
 GE3    -7.665189e+03  -4.054446e-05  1.429e-05  1.743e+00
 GE4    -7.665189e+03  -7.268360e-08  1.432e-06  1.712e+00
 GE5    -7.665189e+03  -1.005413e-09  2.183e-07  1.662e+00
 GE6    -7.665189e+03  1.082752e-11   3.069e-08  1.764e+00
 GE7    -7.665189e+03  -1.392110e-11  2.569e-09  1.739e+00
 GE8    -7.665189e+03  -9.280733e-12  2.863e-11  1.707e+00
```
Only have 25% performance update, which is not enough for SCF with wavefunction file from the same SCF.

Other three test is NOT PASSED. 
for ABACUS-LCAO and `out_wfc_lcao 2`
```
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                         NOTICE
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

 Can't find the wave function file: LOWF.dat
 CHECK IN FILE : OUT.MgO_INITWFC/warning.log

 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                         NOTICE
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

for ABACUS-PW , bad termination will directly arise. 

For VASP. the result in the first SCF is 
```
 entering main loop
       N       E                     dE             d eps       ncg     rms          rms(c)
DAV:   1     0.141448483977E+03    0.14145E+03   -0.14340E+04   496   0.107E+03
DAV:   2    -0.451811018098E+02   -0.18663E+03   -0.17812E+03   688   0.243E+02
DAV:   3    -0.538770348968E+02   -0.86959E+01   -0.86611E+01   616   0.558E+01
DAV:   4    -0.539486845773E+02   -0.71650E-01   -0.71633E-01   560   0.578E+00
DAV:   5    -0.539497453922E+02   -0.10608E-02   -0.10608E-02   576   0.717E-01
DAV:   6    -0.539497688524E+02   -0.23460E-04   -0.23460E-04   568   0.108E-01
DAV:   7    -0.539497693568E+02   -0.50445E-06   -0.50465E-06   472   0.172E-02
DAV:   8    -0.539497694140E+02   -0.57181E-07   -0.56888E-07   352   0.509E-03
DAV:   9    -0.539497694294E+02   -0.15388E-07   -0.15378E-07   264   0.272E-03
DAV:  10    -0.539497694337E+02   -0.43091E-08   -0.43746E-08   264   0.135E-03
DAV:  11    -0.539497694348E+02   -0.11258E-08   -0.12514E-08   248   0.768E-04    0.142E+01
RMM:  12    -0.478312338102E+02    0.61185E+01   -0.20717E+01   490   0.238E+01    0.415E+00
RMM:  13    -0.478705057257E+02   -0.39272E-01   -0.87433E-01   522   0.449E+00    0.273E+00
RMM:  14    -0.478615810209E+02    0.89247E-02   -0.16274E-01   493   0.364E+00    0.560E-01
RMM:  15    -0.478564577104E+02    0.51233E-02   -0.15858E-02   543   0.673E-01    0.338E-01
RMM:  16    -0.478557125188E+02    0.74519E-03   -0.19943E-02   540   0.719E-01    0.118E-01
RMM:  17    -0.478561362735E+02   -0.42375E-03   -0.13903E-03   552   0.193E-01    0.106E-01
RMM:  18    -0.478559822190E+02    0.15405E-03   -0.22735E-04   565   0.898E-02    0.532E-02
RMM:  19    -0.478559373260E+02    0.44893E-04   -0.30012E-04   514   0.988E-02    0.392E-03
RMM:  20    -0.478559386744E+02   -0.13485E-05   -0.15103E-05   411   0.257E-02
   1 F= -.47855939E+02 E0= -.47855939E+02  d E =-.421962E-41
 writing wavefunctions
```
and result in the next SCF:
```
 entering main loop
       N       E                     dE             d eps       ncg     rms          rms(c)
DAV:   1    -0.478559390698E+02   -0.47856E+02   -0.75915E-06   384   0.173E-02
DAV:   2    -0.478559391188E+02   -0.48960E-07   -0.49001E-07   256   0.450E-03
DAV:   3    -0.478559391280E+02   -0.91993E-08   -0.90752E-08   240   0.180E-03
DAV:   4    -0.478559391303E+02   -0.23433E-08   -0.25336E-08   240   0.900E-04
DAV:   5    -0.478559391311E+02   -0.84509E-09   -0.77776E-09   240   0.502E-04
DAV:   6    -0.478559391314E+02   -0.27936E-09   -0.24689E-09   240   0.271E-04
DAV:   7    -0.478559391315E+02   -0.11222E-09   -0.79229E-10   240   0.158E-04
DAV:   8    -0.478559391316E+02   -0.78970E-10   -0.25678E-10   240   0.860E-05
DAV:   9    -0.478559391317E+02   -0.10210E-09   -0.83370E-11   240   0.506E-05
DAV:  10    -0.478559391315E+02    0.24927E-09   -0.27008E-11   240   0.277E-05
DAV:  11    -0.478559391314E+02    0.28010E-10   -0.89275E-12   240   0.164E-05    0.525E-03
RMM:  12    -0.478559385595E+02    0.57194E-06   -0.24059E-06   234   0.843E-03
   1 F= -.47855939E+02 E0= -.47855939E+02  d E =-.453973E-41
```
which have much better performance. 

### Expected behavior

The output wavefunction file should be used for accelerate next SCF calculation,
which will be very useful for ASE-ABACUS to do relax, NEB and other calculation.

### To Reproduce

Do calculation from the input offered. by:
1. copy INPUT_noinit to INPUT. and do SCF calculation
2. copy INPUT_initwfc to INPUT, and do next SCF calcualtion

### Environment

- ABACUS: v3.4.1 Commit: 77f178d04 (Thu Oct 26 11:32:59 2023 +0800) 
- Dependence: Intel-oneAPI toolchain

### Additional Context

As #2975 , ABACUS cannot automatically detect whether wavefunction file is exist, use wavefunction file as guess for SCF calculation if exist, and switch back to default `atomic` initial guess method when setting `init_wfc file` open ( or let automatically detect wavefunction file be default setting, just like VASP do for ISTART setting ). 

I consider that this function should NOT be hard and should be very USEFUL for outer interfaces.

### Task list for Issue attackers (only for developers)

- [X] Verify the issue is not a duplicate.
- [X] Describe the bug.
- [X] Steps to reproduce.
- [X] Expected behavior.
- [X] Error message.
- [X] Environment details.
- [X] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by QuantumMisaka** [comment]
The Problem for reading wfc files from `out_wfc_lcao 1` result is fixed by #3140 , but the binary wfc file is still no use

**Comment by QuantumMisaka** [comment]
Will be partly fixed by #3140 , `out_wfc_lcao 1` and `init_wfc file` can cooperate well

**Comment by QuantumMisaka** [comment]
binary `wfc` usage is not in planned now, this issue may close for now. `out_wfc_lcao 1` can now be cooperated with `init_wfc file`
